src/twisted/trial/test/test_tests.py

```diff
@@ -1453,3 +1490,40 @@ def test_generator(self):
             "is a generator function and therefore will never run",
             result.errors[0][1].value.args[0],
         )
+
+    def test_synchronousTestCaseErrorOnAsyncGeneratorFunction(self):
+        """
+        In a SynchronousTestCase, a test method which is a generator function
+        is reported as an error, as such a method will never run assertions.
```

Comment:
Do we do this for other return value types (i.e. `Deferred`?)  Should we?

Reply:
Yeah Generators

Reply:
Trial doesn't check return types, only the function's flags

Reply:
pytest does this: https://github.com/pytest-dev/pytest/blob/1ba5b485650d6f4ef8ef1ab6e2f782e2cf315ca8/src/_pytest/python.py#L181-L187